src/google-auth.js

```diff
@@ -275,3 +275,34 @@ GoogleSocialProvider.prototype.getAccessTokenFromRefreshToken_ =
     xhr.send({string: data});
   }.bind(this));
 };
+
+// returns Promise<accessToken>
+GoogleSocialProvider.prototype.refreshTokenIfNeeded_ = function(accessToken) {
+  // core.xhr callbacks clobber the xhr object if we use .bind(this), so
+  // instead we should set this to a local variable.
+  var thisSocialProvider = this;
+
+  return new Promise(function(fulfill, reject) {
+    var xhr = freedom["core.xhr"]();
+    var url = 'https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=' +
```

Comment:
In this case we are just checking if the access token is still valid, so we do want to hardcode "oauth2/v1/tokeninfo" as the endpoint